### DISCUSSION TITLE
dApp development

### DISCUSSION THREAD

**Discussion by KK-yao** [zkSync-Community-Hub/zksync-developers#369] [SDKs] [answered]
### Environment

Testnet

### Acknowledgement

- [ ] I understand that issues are expected with the Go SDK as they are being updated.

### Issue Description

In the dApp development tutorial：
singer = await new BrowserProvider(window.ethereum).getSinger();
window.ethereum error report:
Argument of type 'MetaMaskInpageProvider | undefined' is not assignable to parameter of type 'Eip1193Provider'.

### Expected Behavior

able to operate normally

### Code Example

const initializeProviderAndSigner = async () => {
  provider = new Provider("https://sepolia.era.zksync.dev");
  // Note that we still need to get the Metamask signer
  signer = await new BrowserProvider(window.ethereum).getSigner();
  contract = new Contract(GREETER_CONTRACT_ADDRESS, GREETER_CONTRACT_ABI, signer);
};

**Comment by dutterbutter** [comment]
This seems to be an issue specific to MetaMask but please also ensure you have zkSync Sepolia added as a network in mm.

**Comment by knowalosbra** [comment]
It seems like you're encountering a type error when trying to initialize the signer using the BrowserProvider in your dApp development tutorial. This error occurs because the window.ethereum object may be undefined or incompatible with the BrowserProvider. To resolve this issue, ensure that the MetaMask extension is installed and unlocked in your browser, and that the window.ethereum object is properly injected. Additionally, consider handling cases where window.ethereum is undefined to prevent errors.

**Comment by uF4No** [comment, marked as answer]
@KK-yao Make sure you're using `zksync-ethers` version 6.0.0 or higher. Also check if before getting the signer, you're connecting metamask with:

```
await window.ethereum
        .request({ method: "eth_requestAccounts" })
        .catch((e) => console.log(e));
```

**Comment by kaycogrisitbran** [comment]
Ensure that the window.ethereum object is properly initialized and available before calling the getSigner() method. Check for any errors or issues with the MetaMask extension that could be preventing the proper initialization of the ethereum object.

**Comment by amandakob** [comment]
Ensure compatibility between the provided BrowserProvider and the window.ethereum object. Check if window.ethereum is properly initialized and if it supports the EIP-1193 standard. If not, handle the error gracefully or provide alternative methods for obtaining the signer.

**Comment by shannonzimmons** [comment]
The user is experiencing an error with the Go SDK while following a dApp development tutorial on Testnet. The error message reports that the argument type 'MetaMaskInpageProvider | undefined' is not assignable to the parameter type 'Eip1193Provider'. The expected behavior is for the code to operate normally. This issue may be related to updates in the Go SDK and further investigation and debugging may be required to resolve it.
